src/us/kbase/workspace/test/database/provenance/OrganizationTest.java

```diff
@@ -21,72 +34,76 @@ public void equals() throws Exception {
+	public void buildWithNullOrEmptyOrgId() throws Exception {
+		for (String nullOrWs : EMPTY_STRINGS_WITH_NULL) {
+			final Organization org1 = Organization.getBuilder(ORG_NAME)
+					.withOrganizationID(nullOrWs)
+					.build();
+			assertThat(INCORRECT_NAME, org1.getOrganizationName(), is(ORG_NAME));
+			assertThat(INCORRECT_ID, org1.getOrganizationID(), is(ES));
+		}
 	}
 
 	@Test
-	public void buildAndOverwriteOrgIdWithNull() throws Exception {
-		final Organization org1 = Organization.getBuilder("The Organisation").withOrganizationID("The:GrandWazoo")
-				.withOrganizationID("    \t\t   \n   ").build();
-		assertThat("incorrect org name", org1.getOrganizationName(), is("The Organisation"));
-		assertThat("incorrect org ID", org1.getOrganizationID(), is(Optional.empty()));
-
-		final Organization org2 = Organization.getBuilder("The Organisation").withOrganizationID("The:GrandWazoo")
-				.withOrganizationID(null).build();
-		assertThat("incorrect org name", org2.getOrganizationName(), is("The Organisation"));
-		assertThat("incorrect org ID", org2.getOrganizationID(), is(Optional.empty()));
+	public void buildAndOverwriteOrgIdWithNullOrEmpty() throws Exception {
+		for (String nullOrWs : EMPTY_STRINGS_WITH_NULL) {
+			final Organization org1 = Organization.getBuilder(ORG_NAME)
+					.withOrganizationID(PID_STRING)
+					.withOrganizationID(nullOrWs)
+					.build();
+			assertThat(INCORRECT_NAME, org1.getOrganizationName(), is(ORG_NAME));
+			assertThat(INCORRECT_ID, org1.getOrganizationID(), is(ES));
+		}
 	}
 
 	@Test
-	public void buildFailWhitespaceOrgName() throws Exception {
-		try {
-			Organization.getBuilder("          ").build();
-			fail("expected exception");
-		} catch (Exception got) {
-			TestCommon.assertExceptionCorrect(got, new NullPointerException("organizationName"));
+	public void buildFailInvalidPID() throws Exception {
+		for (String invalidPid : INVALID_PID_LIST) {
+			try {
+				Organization.getBuilder(ORG_NAME)
+						.withOrganizationID(invalidPid)
+						.build();
+				fail(EXP_EXC);
+			} catch (Exception got) {
+				TestCommon.assertExceptionCorrect(got, new IllegalArgumentException(
+						"Illegal ID format for organizationID: \"" + invalidPid + "\"\n" +
+								"PIDs should match the pattern \"^([a-zA-Z0-9][a-zA-Z0-9\\.]+)\\s*:\\s*(\\S.+)$\""));
+			}
 		}
 	}
 
 	@Test
-	public void buildFailNullOrgName() throws Exception {
-		try {
-			Organization.getBuilder(null).build();
-			fail("expected exception");
-		} catch (Exception got) {
-			TestCommon.assertExceptionCorrect(got, new NullPointerException("organizationName"));
+	public void buildFailNullOrEmptyOrgName() throws Exception {
+		for (String nullOrWs : EMPTY_STRINGS_WITH_NULL) {
+			try {
+				Organization.getBuilder(nullOrWs).build();
+				fail(EXP_EXC);
+			} catch (Exception got) {
+				TestCommon.assertExceptionCorrect(got, new NullPointerException("organizationName"));
```

Comment:
I must have missed this before, but a null pointer exception shouldn't be thrown if the input is an empty string, only if it's y'know, null

Reply:
Wrote the code that generates that before I knew about `Util.checkString`. Updating it now...

Reply:
:+1: